### PR TITLE
Extension: Display history of conversation + add public api url to list convos

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

- Add a public API route to list convo. It returns a list of `ConversationWithoutContentType`. 
- Extension: create a hook to fetch convos. 
- Extension: display the convos using this hook. 

I need to figure out how to update our api doc, I ran `npm run docs` but did not change anything, will dig. Can be reviewed already. 

<kbd>
Screenshot 2024-10-30 at 17 19 59
</kbd>

## Risk

Can be rolled back. 

## Deploy Plan

Deploy front to get the new API route.

### CODE REVIEW

File: front/pages/api/v1/w/[wId]/assistant/conversations/index.ts [hunk 1]
```diff
@@ -260,13 +269,18 @@ async function handler(
         contentFragment: newContentFragment ?? undefined,
       });
       return;
+    case "GET":
+      const conversations = await getUserConversations(auth);
+      res.status(200).json({ conversations });
+      return;
```
Comment (tdraier) [inline front/pages/api/v1/w/[wId]/assistant/conversations/index.ts:275]: Should we update the swagger doc ?

Comment (PopDaph) [reply front/pages/api/v1/w/[wId]/assistant/conversations/index.ts:275]: Yes I said it in OP but running npm run docs did not change anything 😬

Comment (PopDaph) [reply front/pages/api/v1/w/[wId]/assistant/conversations/index.ts:275]: ok I need to update manually the swagger doc and create the new schema for conversationwithoutcontent

**Review by tdraier — APPROVED** [review]
once swagger doc is done / swagger.json updated

File: front/pages/api/v1/w/[wId]/assistant/conversations/index.ts (hunk 1 above)
Comment (PopDaph) [reply front/pages/api/v1/w/[wId]/assistant/conversations/index.ts:275]: Discussed IRL we don't necessarily want to comm now on it -> need to decide how we expose the ConversationWithoutContent, not a big deal but not blocking and we want to have an MVP 😬

**Comment by PopDaph** [comment]
Created a task assigned to me to not forget about it: https://github.com/dust-tt/tasks/issues/1568

**Pull request merged**